mypyc/test-data/run-classes.test

```diff
@@ -1924,3 +1946,198 @@ from native import A, B, C
 a = A()
 b = B()
 c = C()
+
+[case testCopyAlwaysDefinedAttributes]
+import copy
+from typing import Union
+
+class A: pass
+
+class C:
+    def __init__(self, n: int = 0) -> None:
+        self.n = n
+        self.s = ""
+        self.t = ("", 0)
+        self.u: Union[str, bytes] = ''
+        self.a = A()
+
+def test_copy() -> None:
+    c1 = C()
+    c1.n = 1
+    c1.s = "x"
+    c2 = copy.copy(c1)
+    assert c2.n == 1
+    assert c2.s == "x"
+    assert c2.t == ("", 0)
+    assert c2.u == ''
+    assert c2.a is c1.a
+
+[case testNonNativeCallsToDunderNewAndInit]
+from typing import Any
+from testutil import assertRaises
+
+count_c = 0
+
+class C:
+    def __init__(self) -> None:
```

Comment:
What if `__init__` calls another method before setting an attribute? For example, insert a call to `self.get()` before assigning to `self.x` here.

Reply:
In that case we can't infer any always defined attributes from `__init__`. The analysis stops the moment any other function (other than base class `__init__`) can see `self`.

Reply:
Oh great, would it make sense to add a test demonstrating that.

Reply:
The `testAlwaysDefinedFail` test case covers this.